src/components/settings/Users/Users.vue

```diff
@@ -0,0 +1,147 @@
+                @click="$router.push({ name: 'settings.users.edit', params: { id } })"
+            >
+                <div class="col-10 col-sm-5 m-2 flex-shrink-1 fw-bold">
+                    {{ username }}
+                </div>
+                <div class="col-5 px-1 flex-shrink-1 d-none d-sm-flex gap-2 align-items-center">
+                    <font-awesome-icon :class="active ? 'text-success' : 'text-muted'" :icon="active ? 'check-circle' : 'times-circle'" />
+                    <div>{{ $t(active ? "Active" : "Inactive") }}</div>
+                </div>
+                <div class="col-2 pe-2 pe-lg-3 d-flex justify-content-end">
+                    <button
+                        type="button"
+                        class="btn-ban-user btn ms-2 py-1"
+                        :class="active ? 'btn-outline-danger' : 'btn-outline-success'"
+                        :disabled="processing"
+                        @click.stop="active ? disableConfirm(index) : toggleActiveUser(index)"
+                    >
+                        <font-awesome-icon class="" :icon="active ? 'user-slash' : 'user-check'" />
+                    </button>
+                </div>
+            </div>
+        </div>
+
+        <Confirm ref="confirmDisable" btn-style="btn-danger" :yes-text="$t('Yes')" :no-text="$t('No')" @yes="toggleActiveUser">
+            {{ $t("confirmDisableUserMsg") }}
+        </Confirm>
+    </div>
+</template>
+
+<script>
+import { useToast } from "vue-toastification";
+import Confirm from "../../Confirm.vue";
+const toast = useToast();
+
+export default {
+    components: { Confirm },
+
+    data: () => ({
+        loading: false,
+        processing: false,
+        usersList: null,
+        disablingUser: null,
+    }),
+
+    mounted() {
+        this.getUsers();
+    },
+
+    methods: {
+        /**
+         * Get list of users from server
+         * @returns {void}
+         */
+        getUsers() {
+            this.loading = true;
+            this.$root.getSocket().emit("getUsers", (res) => {
+                this.loading = false;
+                if (res.ok) {
+                    this.usersList = res.users;
+                } else {
+                    toast.error(res.msg);
+                }
+            });
+        },
+
+        /**
+         * Show confirmation for disabling a user
+         * @param {number} index index of the user to disable in the local usersList
+         * @returns {void}
+         */
+        disableConfirm(index) {
+            this.disablingUser = this.usersList[index];
+            this.$refs.confirmDisable.show();
+        },
+
+        /**
+         * Disable a user from server
+         * @param {number} [index] index of the user to disable in the local usersList, use disablingUser if not provided
+         * @returns {void}
+         */
+        toggleActiveUser(index) {
+            const { active, ...rest } = this.usersList[index] || this.disablingUser;
```

Comment:
Because `index` is not always "passed anyway" 😜 

Check L. 41: the same function is used after the user click on "yes" in the confirm modal, but the confirm modal is global for the whole list and so, haven't access to the `index`. In this case, the function is called without the `index` and the `disablingUser` is used instead.

I'm a bit surprise by this comment because I try to explicit the usage with JSDoc. Have you a suggestion to enhance the doc?

Reply:
This still leaves two ways of calling this function with a user, which is not really clean
⇒ I would prefer a function to only consider one side effect at most.

Reply:
Solved in another way, see last commit